classes/requests/class-kp-requests.php

```diff
@@ -99,7 +99,7 @@ public function set_credentials() {
 	}
 
 	public function calculate_auth() {
-		return 'Basic ' . base64_encode( $this->merchant_id . ':' . htmlspecialchars_decode( $this->shared_secret ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions -- Base64 used to calculate auth headers.
+		return 'Basic ' . base64_encode( $this->merchant_id . ':' . htmlspecialchars_decode( $this->shared_secret, ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401 ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions -- Base64 used to calculate auth headers.
```

Comment:
`htmlspecialchars_decode` default flags are already  `ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401`. So why is that needed here?
https://www.php.net/manual/en/function.htmlspecialchars-decode.php

Reply:
That's correct, however it is required to be explicitly set.

![image](https://github.com/user-attachments/assets/e531f230-adb2-404b-854d-4c1c614cd257)

Reply:
Ah ok, then i see. Thanks for the clarification